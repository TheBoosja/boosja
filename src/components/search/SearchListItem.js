import React from 'react';
import moment from 'moment';
import { ResultItem, ResultImage, ResultName, QuaternaryHeading } from 'components/styled';

const SearchListItem = ({ show: { id, poster_path, name, first_air_date } }) => {
	const source = poster_path ? `https://image.tmdb.org/t/p/w300${poster_path}` : 'none';

	const firstAirDate = first_air_date ? moment(first_air_date).format('YYYY') : 'N/A';

	return (
		<ResultItem to={`/tv/${id}`}>
			<ResultImage src={source} alt={`${name} poster`} />
			<ResultName>
				<QuaternaryHeading>{name}</QuaternaryHeading>
				<p>{firstAirDate}</p>
			</ResultName>
		</ResultItem>
	);
};

export default SearchListItem;
