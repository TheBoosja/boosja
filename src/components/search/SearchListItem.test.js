import React from 'react';
import { shallow } from 'enzyme';
import SearchListItem from './SearchListItem';

describe('Search List Item', () => {
	const show = { name: 'Test' };
	const props = { show };
	const listItem = shallow(<SearchListItem {...props} />);

	it('renders properly', () => {
		expect(listItem).toMatchSnapshot();
	});

	it('contains a ResultItem component', () => {
		expect(listItem.find('ResultItem').exists()).toBe(true);
	});
});
