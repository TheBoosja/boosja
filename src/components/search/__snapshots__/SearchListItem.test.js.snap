// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search List Item renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchListItem
    show={
        Object {
            "name": "Test",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ResultImage
          alt="Test poster"
          src="none"
/>,
        <ResultName>
          <QuaternaryHeading>
                    Test
          </QuaternaryHeading>
          <p>
                    N/A
          </p>
</ResultName>,
      ],
      "to": "/tv/undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "Test poster",
          "src": "none",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <QuaternaryHeading>
              Test
</QuaternaryHeading>,
            <p>
              N/A
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "N/A",
            },
            "ref": null,
            "rendered": "N/A",
            "type": "p",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ResultImage
            alt="Test poster"
            src="none"
/>,
          <ResultName>
            <QuaternaryHeading>
                        Test
            </QuaternaryHeading>
            <p>
                        N/A
            </p>
</ResultName>,
        ],
        "to": "/tv/undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "Test poster",
            "src": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <QuaternaryHeading>
                Test
</QuaternaryHeading>,
              <p>
                N/A
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test",
              },
              "ref": null,
              "rendered": "Test",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "N/A",
              },
              "ref": null,
              "rendered": "N/A",
              "type": "p",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
